packages/orchestrator/internal/template/metadata/template_metadata.go

```diff
@@ -5,83 +5,139 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
-	"path"
+	"io"
+	"os"
 
-	"github.com/e2b-dev/infra/packages/orchestrator/internal/template/build/sandboxtools"
+	"github.com/e2b-dev/infra/packages/shared/pkg/ioutils"
 	"github.com/e2b-dev/infra/packages/shared/pkg/storage"
 )
 
 const (
-	metadataVersion      = "v1"
-	templateMetadataFile = "metadata.json"
+	CurrentVersion = 2
 )
 
-type FromTemplateMetadata struct {
+type Command struct {
+	User    string            `json:"user,omitempty"`
+	WorkDir *string           `json:"workdir,omitempty"`
+	EnvVars map[string]string `json:"env_vars,omitempty"`
+}
+
+type FromTemplate struct {
 	Alias   string `json:"alias"`
 	BuildID string `json:"build_id"`
 }
 
-type StartMetadata struct {
-	StartCmd string                       `json:"start_command"`
-	ReadyCmd string                       `json:"ready_command"`
-	Metadata sandboxtools.CommandMetadata `json:"metadata"`
+type Start struct {
+	StartCmd string  `json:"start_command"`
+	ReadyCmd string  `json:"ready_command"`
+	Metadata Command `json:"metadata"`
 }
 
-type TemplateMetadata struct {
-	Version      string                       `json:"version"`
-	Template     storage.TemplateFiles        `json:"template"`
-	Metadata     sandboxtools.CommandMetadata `json:"metadata"`
-	Start        *StartMetadata               `json:"start,omitempty"`
-	FromImage    *string                      `json:"from_image,omitempty"`
-	FromTemplate *FromTemplateMetadata        `json:"from_template,omitempty"`
+type Template struct {
+	Version      uint64                `json:"version"`
+	Template     storage.TemplateFiles `json:"template"`
+	Metadata     Command               `json:"metadata"`
```

Comment:
this is little bit weird, as you have in `LayerResult` field 
```
	Metadata metadata.Template
```

and then `Template` also has `Metadata`